vulnerabilities/importer.py

```diff
@@ -444,6 +502,8 @@ class AdvisoryData:
     date_published: Optional[datetime.datetime] = None
     weaknesses: List[int] = dataclasses.field(default_factory=list)
     severities: List[VulnerabilitySeverity] = dataclasses.field(default_factory=list)
+    fixed_by_commits: List[CodeCommitData] = dataclasses.field(default_factory=list)
+    affected_by_commits: List[CodeCommitData] = dataclasses.field(default_factory=list)
```

Comment:
This is not correct. `affected_by_commits` and `fixed_by_commit` should be part of `AffectedPackage`, not directly part of `AdvisoryData`. In our django model we keep the affected and fixed commits in `ImpactedPackage` and `AffectedPackage` is a low level representation of that same `ImpactedPackage`.

That's the only way to properly capture which version/commit range is fixed by which version/commit. An advisory can have multiple such sets of affected and fixed versions/commits.

Reply:
@keshav-space Yes, I think it should be part of AffectedPackage, but this would force us to create an ImpactedPackage relation before we can store fix commits. For example, if we have a pipeline that loops over 1,000 Git repositories to extract fixed commits, we might not be able to get ImpactedPackages (because there’s no purl provided for AffectedPackage).

Therefore, I think `fixed_by_commits` and `affected_by_commits` should be standalone if we don’t have ImpactedPackage/AffectedPackage.

Not sure but I can reimplement this if needed, I thought we have a previous discussion on this @TG1999

Reply:
@ziadhany  

> For example, if we have a pipeline that loops over 1,000 Git repositories to extract fixed commits, we might not be able to get ImpactedPackages (because there’s no purl provided for AffectedPackage).

Don't you have vcs url for this commit? we can create purl from vcs url. Can you share some example.

Reply:
Then how will we store vcs URL, and commit hashes ? Since current affectedpackage data class does not support any of this ?

Reply:
we're going with the second one.